src/client.rs

```diff
@@ -556,14 +671,32 @@ impl QdrantClient {
         let collection_name = collection_name.to_string();
         let collection_name_ref = collection_name.as_str();
         let ordering_ref = ordering.as_ref();
-
         Ok(self
             .with_points_client(|mut points_api| async move {
+                if points.len() > 1024 {
```

Comment:
TBH I am on the fence on this one :thinking: 

It is modifying an existing function with a new behavior that may be surprising users.

If it were a completely new function with clear splitting semantic I'd be all onboard.

Happy to hear more opinions!

Reply:
Andrey told me the python client is already doing this, and the only visible change in behavior is that the user doesn't get a timeout when upserting too many points at once. For upserting less than 1025 points, the behavior is exactly the same as before.

Reply:
> Andrey told me the python client is already doing this

No, python client is doing it in a separate function. 

I would agree with @agourlay, that this changing existing function might be surprising and inconsistent with all other APIs we have.  

Batch size of 1024 looks arbitrary: it doesn't guarantee no timeouts, single vector can have huge payload, or, on the other hand, vectors can be very small, in this case 1024 is not that big of a batch size. Python client allows user to decide on this, and it can do it only because of the dedicated function for batched uplaod

Reply:
I see. So we could either allow using a heuristic depending on vector + payload size or we let the user decide how big to make the batch.

Reply:
> let the user decide how big to make the batch.

This would change the interface. That's why I would prefer a dedicated function.

Reply:
Ok, I'm convinced. I'll add the `_batch` versions to both blocking and non-blocking variants.